.github/workflows/test-deploy.yml

```
@@ -27,6 +32,6 @@ jobs:
       - name: Deploy to PyPI
         run: poetry publish
         continue-on-error: true
-        if: success() && github.event_name == 'push' && github.ref == 'refs/heads/master'
+        if: success() && github.event_name == 'release'
```

You could add further conditions, but unless there are tags you wish to whitelist or blacklist it comes down to the same thing. See https://help.github.com/en/actions/reference/contexts-and-expression-syntax-for-github-actions#functions
```suggestion
        if: success() && github.event_name == 'release' && startsWith(github.ref, 'refs/tags/')
```